### ISSUE TITLE
BUG: tf filter trailing and leading zeros in residuez

### ISSUE THREAD

**Issue by endolith** [scipy/scipy#4561]
Originally in https://github.com/scipy/scipy/issues/4464, a bug I noticed:

```
residue([6, 0], [1, 5, 8, 4])
Out[33]: (array([  6.,  12.,  -6.]), array([-2., -2., -1.]), array([ 0.]))

residue([0, 0, 6, 0], [1, 5, 8, 4])
Out[34]: (array([  6.,  12.,  -6.]), array([-2., -2., -1.]), array([ 0.]))

residue([0, 0, 6, 0], [0, 0, 0, 1, 5, 8, 4])
Out[36]: (array([ inf,  inf, -inf]), array([-2., -2., -1.]), array([ 0.]))
```

These should all be identical.  Same for these:

```
residuez([6], [1, 5, 8, 4])
Out[37]: (array([-12.,  12.,   6.]), array([-2., -2., -1.]), array([ 0.]))

residuez([6, 0, 0], [1, 5, 8, 4])
Out[38]: (array([-12.,  12.,   6.]), array([-2., -2., -1.]), array([ 0.]))

residuez([6, 0, 0], [1, 5, 8, 4, 0, 0])
Out[39]:
(array([-12.,  12.,   6.,  nan,  nan]),
 array([-2., -2., -1.,  0.,  0.]),
 array([ 0.]))
```

```
invresz(array([-9.,  3.,  6.]), array([-2., -2., -1.]), [])
Out[55]: (array([ 0.,  0.,  6.]), array([ 1.,  5.,  8.,  4.]))

invresz(array([-9.,  3.,  6.]), array([-2., -2., -1.]), [0])
Out[56]: (array([ 0.,  0.,  6.,  0.]), array([ 1.,  5.,  8.,  4.]))
```

For negative powers tf like in `residuez` and `invresz`, trailing zeros don't matter, so shouldn't be returned.  `[6, 0, 0]` represents 6 + 0_z^-1 + 0_z^-2 = `[6]`, for instance.

For positive powers tf like in `residue`, `invres`, and `tf2__`, leading zeros don't matter. `[0, 0, 6, 0]` represents 0_s^3 + 0_s^2 + 6*s + 0 = `[6, 0]`, for instance.

invres seems ok:

```
invres(array([  6.,  12.,  -6.]), array([-2., -2., -1.]), array([ 0.]))
Out[71]: (array([ 6.,  0.]), array([ 1.,  5.,  8.,  4.]))

invres(array([  6.,  12.,  -6.]), array([-2., -2., -1.]), array([ ]))
Out[72]: (array([ 6.,  0.]), array([ 1.,  5.,  8.,  4.]))

invres(array([  6.,  12.,  -6.]), array([-2., -2., -1.]), array([0, 0]))
Out[73]: (array([ 6.,  0.]), array([ 1.,  5.,  8.,  4.]))
```

Other functions that input or output `tf` format should also be checked.

**Comment by larsoner** [comment]
I can confirm that the `residue` / `residuez` parts have been fixed by #10997, keeping this open because of the `invresz` problem(s)

**Comment by nmayorov** [comment]
Let me comment about leading and trailing zeros in `invres` and `invresz` results.

If we have `n` poles and the length of direct term is `m`, then the numerator has length `n + m` and the denominator has length `n + 1`. 

As we doing numerical computations we can't check for trailing or leading zeros in the result. We can only check for exact zeros in the input `k`. 

In the example `invres([ 6,  12,  -6], [-2, -2, -1], [0])` the correct (expected) answer is `b = [0, 6, 0], a = [1, 5, 8, 4]`. This is the result of MATLAB. The zero in `k = [0]` was removed, but the leading numerical zero in `b` is not removed.
